### PR TITLE
Added a post-processor to compute relative difference between 2 post-processors

### LINKED ISSUE #10148: Add a post-processor to compute the relative difference between 2 post-processors
## Rationale
It is useful to be to able to test that 2 post-processor values are relatively equal; this can be achieved by computing the relative difference between them and comparing to a tolerance. For instance, a test might output this relative difference value to a CSV file and then use the `CSVDiff` tester with `abs_zero = my_relative_tol` and `rel_tol = 0`.

## Description
This adds a post-processor that computes the relative difference between 2 post-processor values.

## Impact
This only adds a new post-processor; it does not affect any existing code.

### PR DESCRIPTION
**Pull request by joshuahansel**
Added a post-processor to compute relative difference between 2 post-processors.

Closes #10148

### CODE REVIEW

**Comment by dschwen** [comment]
Linear combination, relative difference... looks like it is starting to make sense to add a `ParsedPostprocessor`

**Comment by permcody** [comment]
Let's just jump right to `ParsedKernel`. In fact, we'll just let you write your weak form in a simple python script...

**Comment by YaqiWang** [comment]
I believe I once required this `ParsedPostprocessor` but ended up with a `ScalePostprocessor`.

File: framework/src/postprocessors/RelativeDifferencePostprocessor.C [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+void
+RelativeDifferencePostprocessor::execute()
+{
+}
+
+PostprocessorValue
+RelativeDifferencePostprocessor::getValue()
+{
+  if (std::abs(_value2) < _zero_tol)
```
Comment (permcody) [inline framework/src/postprocessors/RelativeDifferencePostprocessor.C]: FYI - We coded this common operation up a long time ago:
https://github.com/idaholab/moose/blob/devel/framework/include/utils/MooseUtils.h#L213

Comment (permcody) [reply framework/src/postprocessors/RelativeDifferencePostprocessor.C]: Oh sorry - that's not the correct method. You are comparing against zero here. There are several methods in that file you might want to check out.

**Comment by dschwen** [comment]
> Let's just jump right to `ParsedKernel`. In fact, we'll just let you write your weak form in a simple python script...

My new function parser will actually make a parsed kernel totally possible! With actual jacobian derivation.

File: framework/src/postprocessors/RelativeDifferencePostprocessor.C (hunk 1 above)
Comment (joshuahansel) [reply framework/src/postprocessors/RelativeDifferencePostprocessor.C]: Ok thanks, I will use `absoluteFuzzyEqual` then.

**Comment by joshuahansel** [comment]
Let me know when Santa brings us a `ParsedPostprocessor`. In the end what I'd really like is a `ParsedTester` that works on post-processor values directly instead of going through an output file...

**Review by andrsd — APPROVED** [review]

**Pull request merged**